From 1260c7496c956c8faf7da0fbbf022fa46297f024 Mon Sep 17 00:00:00 2001
From: Zeng Heng <zengheng4@huawei.com>
Date: Mon, 22 Jan 2024 14:58:29 +0800
Subject: [PATCH 677/960] downstream: arm_mpam: control memory bandwidth with
 hard limit flag

When the max memory bandwidth is exceeded, the partition does not use any
more bandwidth until the memory bandwidth measurement for the partition
falls below max value setting.

Signed-off-by: Zeng Heng <zengheng4@huawei.com>
Signed-off-by: Liu Shixin <liushixin2@huawei.com>
---
 drivers/platform/mpam/mpam_devices.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/platform/mpam/mpam_devices.c b/drivers/platform/mpam/mpam_devices.c
index 688b5018e4cc..970a279ae780 100644
--- a/drivers/platform/mpam/mpam_devices.c
+++ b/drivers/platform/mpam/mpam_devices.c
@@ -1189,7 +1189,8 @@ static void mpam_reprogram_ris_partid(struct mpam_msc_ris *ris, u16 partid,
 
 	if (mpam_has_feature(mpam_feat_mbw_max, rprops)) {
 		if (mpam_has_feature(mpam_feat_mbw_max, cfg))
-			mpam_write_partsel_reg(msc, MBW_MAX, cfg->mbw_max);
+			mpam_write_partsel_reg(msc, MBW_MAX, cfg->mbw_max |
+									MPAMCFG_MBW_MAX_HARDLIM);
 		else
 			mpam_write_partsel_reg(msc, MBW_MAX, bwa_fract);
 	}
-- 
2.43.0

